onnxruntime/core/providers/cpu/tensor/scatter_nd.cc

```diff
@@ -11,73 +12,87 @@ ONNX_CPU_OPERATOR_VERSIONED_KERNEL(
     11,
     12,
     KernelDefBuilder()
-        .TypeConstraint("T", DataTypeImpl::AllTensorTypes())
-        .TypeConstraint("Tind", DataTypeImpl::GetTensorType<int64_t>()),
+        .TypeConstraint("T", DataTypeImpl::AllTensorTypes()),
     ScatterND);
 
 ONNX_CPU_OPERATOR_KERNEL(
     ScatterND,
     13,
     KernelDefBuilder()
-        .TypeConstraint("T", DataTypeImpl::AllTensorTypes())
-        .TypeConstraint("Tind", DataTypeImpl::GetTensorType<int64_t>()),
+        .TypeConstraint("T", DataTypeImpl::AllTensorTypes()),
     ScatterND);
 
-template <typename Tind>
-Status ScatterNDBase::PrepareForCompute(OpKernelContext* context, Prepare& p) const {
-  const auto* input_tensor = context->Input<Tensor>(0);
-  const auto* indice_tensor = context->Input<Tensor>(1);
-  const auto* update_tensor = context->Input<Tensor>(2);
-  ORT_ENFORCE(input_tensor != nullptr);
-  ORT_ENFORCE(indice_tensor != nullptr);
-  ORT_ENFORCE(update_tensor != nullptr);
+Status ScatterNDBase::ValidateShapes(const TensorShape& input_shape,
```

Comment:
Does this function need to be a member function?

Reply:
It needn’t - can make it static

Reply:
Fixed